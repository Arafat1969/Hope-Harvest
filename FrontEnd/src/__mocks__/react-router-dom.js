// src/__mocks__/react-router-dom.js
const React = require("react");

function makeAnchor({ children, to, className, ...rest }) {
  // ensure className is a string
  const cn = Array.isArray(className) ? className.join(" ") : className;
  return (
    <a href={to} className={cn} {...rest}>
      {children}
    </a>
  );
}

module.exports = {
  BrowserRouter: ({ children }) => <>{children}</>,
  MemoryRouter: ({ children }) => <>{children}</>,
  Routes: ({ children }) => <>{children}</>,
  Route: ({ element }) => element,

  Link: (props) => makeAnchor(props),
  NavLink: (props) => makeAnchor(props),

  useNavigate: () => () => {},
  useLocation: () => ({ pathname: "/" }),
};
