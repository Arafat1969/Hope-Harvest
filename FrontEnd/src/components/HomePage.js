import { Link } from 'react-router-dom';

const styles = {
  heroSection: {
    background: 'linear-gradient(135deg, #4CAF50 0%, #2E7D32 100%)',
    color: 'white',
    minHeight: '500px',
    display: 'flex',
    alignItems: 'center'
  },
  featureCard: {
    borderRadius: '0.75rem',
    boxShadow: '0 4px 6px rgba(0, 0, 0, 0.1)',
    transition: 'transform 0.3s, box-shadow 0.3s',
    border: 'none'
  },
  ctaButton: {
    backgroundColor: '#2E7D32',
    borderColor: '#2E7D32',
    borderRadius: '2rem',
    padding: '12px 30px',
    fontWeight: '600'
  },
  statCard: {
    backgroundColor: '#f8f9fa',
    borderRadius: '0.75rem',
    padding: '2rem',
    textAlign: 'center'
  }
};

const HomePage = () => {
  return (
    <div>
      {/* Hero Section */}
      <section style={styles.heroSection}>
        <div className="container">
          <div className="row align-items-center">
            <div className="col-lg-6">
              <h1 className="display-4 fw-bold mb-4">
                Hope Harvest
              </h1>
              <h2 className="h3 mb-4">
                Building Sustainable Communities Through Compassion
              </h2>
              <p className="lead mb-4">
                Join us in creating lasting change through microfinance, relief distribution, 
                education, and community development programs across Bangladesh.
              </p>
              <div className="d-flex gap-3">
                <Link to="/campaigns">
                  <button className="btn btn-light btn-lg" style={styles.ctaButton}>
                    <i className="fas fa-heart me-2"></i>
                    View Campaigns
                  </button>
                </Link>
                <Link to="/donate">
                  <button className="btn btn-outline-light btn-lg">
                    <i className="fas fa-hand-holding-heart me-2"></i>
                    Donate Now
                  </button>
                </Link>
              </div>
            </div>
            <div className="col-lg-6 text-center">
              <div className="p-4">
                <i className="fas fa-seedling fa-5x mb-3" style={{ opacity: 0.8 }}></i>
                <h4>Growing Hope Together</h4>
              </div>
            </div>
          </div>
        </div>
      </section>

      {/* Impact Statistics */}
      <section className="py-5 bg-light">
        <div className="container">
          <div className="row text-center">
            <div className="col-md-3 mb-4">
              <div style={styles.statCard}>
                <i className="fas fa-users fa-3x text-success mb-3"></i>
                <h3 className="text-success">10,000+</h3>
                <p className="text-muted">Lives Impacted</p>
              </div>
            </div>
            <div className="col-md-3 mb-4">
              <div style={styles.statCard}>
                <i className="fas fa-bullhorn fa-3x text-success mb-3"></i>
                <h3 className="text-success">150+</h3>
                <p className="text-muted">Active Campaigns</p>
              </div>
            </div>
            <div className="col-md-3 mb-4">
              <div style={styles.statCard}>
                <i className="fas fa-hand-holding-heart fa-3x text-success mb-3"></i>
                <h3 className="text-success">à§³5M+</h3>
                <p className="text-muted">Funds Raised</p>
              </div>
            </div>
            <div className="col-md-3 mb-4">
              <div style={styles.statCard}>
                <i className="fas fa-hands-helping fa-3x text-success mb-3"></i>
                <h3 className="text-success">500+</h3>
                <p className="text-muted">Volunteers</p>
              </div>
            </div>
          </div>
        </div>
      </section>

      {/* Key Features */}
      <section className="py-5">
        <div className="container">
          <div className="text-center mb-5">
            <h2 className="display-5 text-success fw-bold">How We Make a Difference</h2>
            <p className="lead text-muted">Our comprehensive approach to community development</p>
          </div>
          
          <div className="row">
            <div className="col-lg-4 mb-4">
              <div 
                className="card h-100" 
                style={styles.featureCard}
                onMouseOver={(e) => {
                  e.currentTarget.style.transform = 'translateY(-10px)';
                  e.currentTarget.style.boxShadow = '0 8px 25px rgba(0,0,0,0.15)';
                }}
                onMouseOut={(e) => {
                  e.currentTarget.style.transform = 'translateY(0)';
                  e.currentTarget.style.boxShadow = '0 4px 6px rgba(0, 0, 0, 0.1)';
                }}
              >
                <div className="card-body text-center p-4">
                  <i className="fas fa-seedling fa-3x text-success mb-3"></i>
                  <h4 className="text-success">Self-Reliance Programs</h4>
                  <p className="text-muted">
                    Microfinance, skills development, and entrepreneurship training 
                    to build sustainable livelihoods.
                  </p>
                  <Link to="/campaigns" className="btn btn-outline-success">
                    Learn More
                  </Link>
                </div>
              </div>
            </div>

            <div className="col-lg-4 mb-4">
              <div 
                className="card h-100" 
                style={styles.featureCard}
                onMouseOver={(e) => {
                  e.currentTarget.style.transform = 'translateY(-10px)';
                  e.currentTarget.style.boxShadow = '0 8px 25px rgba(0,0,0,0.15)';
                }}
                onMouseOut={(e) => {
                  e.currentTarget.style.transform = 'translateY(0)';
                  e.currentTarget.style.boxShadow = '0 4px 6px rgba(0, 0, 0, 0.1)';
                }}
              >
                <div className="card-body text-center p-4">
                  <i className="fas fa-hands-helping fa-3x text-success mb-3"></i>
                  <h4 className="text-success">Emergency Relief</h4>
                  <p className="text-muted">
                    Rapid response to natural disasters, food security, 
                    and essential support for vulnerable communities.
                  </p>
                  <Link to="/campaigns" className="btn btn-outline-success">
                    Learn More
                  </Link>
                </div>
              </div>
            </div>

            <div className="col-lg-4 mb-4">
              <div 
                className="card h-100" 
                style={styles.featureCard}
                onMouseOver={(e) => {
                  e.currentTarget.style.transform = 'translateY(-10px)';
                  e.currentTarget.style.boxShadow = '0 8px 25px rgba(0,0,0,0.15)';
                }}
                onMouseOut={(e) => {
                  e.currentTarget.style.transform = 'translateY(0)';
                  e.currentTarget.style.boxShadow = '0 4px 6px rgba(0, 0, 0, 0.1)';
                }}
              >
                <div className="card-body text-center p-4">
                  <i className="fas fa-graduation-cap fa-3x text-success mb-3"></i>
                  <h4 className="text-success">Education & Training</h4>
                  <p className="text-muted">
                    Comprehensive education programs, vocational training, 
                    and digital literacy initiatives.
                  </p>
                  <Link to="/campaigns" className="btn btn-outline-success">
                    Learn More
                  </Link>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>

      {/* Call to Action */}
      <section className="py-5 bg-success text-white">
        <div className="container text-center">
          <h2 className="display-5 fw-bold mb-4">Ready to Make a Difference?</h2>
          <p className="lead mb-4">
            Join thousands of donors and volunteers who are building a better future for Bangladesh
          </p>
          <div className="d-flex gap-3 justify-content-center flex-wrap">
            <Link to="/donate">
              <button className="btn btn-light btn-lg">
                <i className="fas fa-heart me-2"></i>
                Start Donating
              </button>
            </Link>
            <Link to="/volunteer-activity">
              <button className="btn btn-outline-light btn-lg">
                <i className="fas fa-hands-helping me-2"></i>
                Become a Volunteer
              </button>
            </Link>
            <Link to="/apply-for-funds">
              <button className="btn btn-outline-light btn-lg">
                <i className="fas fa-bullhorn me-2"></i>
                Start a Campaign
              </button>
            </Link>
          </div>
        </div>
      </section>
    </div>
  );
};

export default HomePage;
