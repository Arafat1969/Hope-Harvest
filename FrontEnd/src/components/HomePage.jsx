import { useState, useEffect, useRef } from 'react';
import { Link } from 'react-router-dom';
import styles from './HomePage.module.css';
import AnimatedNumber from '../components/AnimatedNumber'; // Adjust path if your components folder is elsewhere

// Custom Hook to detect when an element is in view
const useInView = (options) => {
  const [isInView, setIsInView] = useState(false);
  const ref = useRef(null);

  useEffect(() => {
    const observer = new IntersectionObserver(([entry]) => {
      if (entry.isIntersecting) {
        setIsInView(true);
        observer.disconnect();
      }
    }, options);

    if (ref.current) {
      observer.observe(ref.current);
    }

    return () => {
      if (ref.current) {
        observer.unobserve(ref.current);
      }
    };
  }, [ref, options]);

  return [ref, isInView];
};


const HomePage = () => {
  const heroImages = [
    'https://upload.wikimedia.org/wikipedia/commons/thumb/c/c9/Agriculture_in_Rural_Bangladesh.jpg/2560px-Agriculture_in_Rural_Bangladesh.jpg',
    'https://tds-images.thedailystar.net/sites/default/files/styles/very_big_201/public/images/2023/03/22/char_1.jpg',
    'https://cdn.pixabay.com/photo/2021/02/12/16/08/children-6008967_1280.jpg',
    'https://s.france24.com/media/display/cb41bca0-6201-11ef-9d3e-005056bf30b7/w:1280/p:16x9/5db8797c814b2f749a6d17aa7a5dc022e7f7be13.jpg',
    'https://im.indiatimes.in/amp/2019/Jul/trees_1562664811.jpg?w=1200&h=900&cc=1&webp=1&q=75'
  ];

  const [currentImageIndex, setCurrentImageIndex] = useState(0);
  const [statsSectionRef, isStatsSectionVisible] = useInView({ threshold: 0.1 });

  useEffect(() => {
    const timer = setInterval(() => {
      setCurrentImageIndex((prevIndex) =>
        (prevIndex + 1) % heroImages.length
      );
    }, 2000);

    return () => clearInterval(timer);
  }, [heroImages.length]);

  return (
    <div>
      {/* Hero Section */}
      <section className={styles.heroSection}>
        <div className={styles.heroBackground}>
          {heroImages.map((src, index) => (
            <img
              key={src}
              src={src}
              alt="Hope Harvest initiative background"
              className={`${styles.backgroundImage} ${index === currentImageIndex ? styles.visible : ''}`}
            />
          ))}
        </div>
        <div className={`container ${styles.heroContent}`}>
          <div className="row">
            <div className="col-lg-7 text-center text-lg-start">
              <h1 className={styles.heroTitle}>Hope Harvest</h1>
              <h2 className={styles.heroSubtitle}>Sowing Seeds of Change, Reaping a Brighter Future.</h2>
              <p className={styles.heroText}>
                We empower communities across Bangladesh through sustainable microfinance, education, and direct relief, creating lasting, positive change from the ground up.
              </p>
              <div className="d-flex gap-3 justify-content-center justify-content-lg-start">
                <Link to="/donate" className={`btn ${styles.ctaButtonPrimary}`}>
                  <i className="fas fa-hand-holding-heart me-2"></i>
                  Donate Now
                </Link>
                <Link to="/campaigns" className={`btn ${styles.ctaButtonSecondary}`}>
                  View Campaigns
                </Link>
              </div>
            </div>
          </div>
        </div>
      </section>

      {/* Impact Statistics Section */}
      <section ref={statsSectionRef} className={styles.statsSection}>
        <div className="container">
          <div className="row g-4">
            <div className="col-lg-3 col-md-6">
              <div className={styles.statCard}>
                <i className={`fas fa-users ${styles.statIcon}`}></i>
                <h3 className={styles.statNumber}>
                  {isStatsSectionVisible ? <AnimatedNumber target="10,000+" /> : '0+'}
                </h3>
                <p className={styles.statLabel}>Lives Impacted</p>
              </div>
            </div>
            <div className="col-lg-3 col-md-6">
              <div className={styles.statCard}>
                <i className={`fas fa-bullhorn ${styles.statIcon}`}></i>
                <h3 className={styles.statNumber}>
                  {isStatsSectionVisible ? <AnimatedNumber target="150+" /> : '0+'}
                </h3>
                <p className={styles.statLabel}>Active Campaigns</p>
              </div>
            </div>
            <div className="col-lg-3 col-md-6">
              <div className={styles.statCard}>
                <i className={`fas fa-hand-holding-heart ${styles.statIcon}`}></i>
                <h3 className={styles.statNumber}>
                   {isStatsSectionVisible ? <AnimatedNumber target="5,000,000+" /> : '0+'}
                </h3>
                <p className={styles.statLabel}>Taka Raised</p>
              </div>
            </div>
            <div className="col-lg-3 col-md-6">
              <div className={styles.statCard}>
                <i className={`fas fa-hands-helping ${styles.statIcon}`}></i>
                <h3 className={styles.statNumber}>
                  {isStatsSectionVisible ? <AnimatedNumber target="500+" /> : '0+'}
                </h3>
                <p className={styles.statLabel}>Volunteers</p>
              </div>
            </div>
          </div>
        </div>
      </section>

      {/* Key Features Section */}
      <section className={styles.featuresSection}>
        <div className="container">
          <div className="text-center mb-5">
            <h2 className={styles.sectionTitle}>How We Make a Difference</h2>
            <p className={styles.sectionSubtitle}>Our comprehensive approach to community development</p>
          </div>
          <div className="row">
            <div className="col-lg-4 mb-4">
              <div className={`card h-100 ${styles.featureCard}`}>
                <div className="card-body p-4">
                  <i className={`fas fa-seedling ${styles.featureIcon}`}></i>
                  <h4 className={styles.featureTitle}>Self-Reliance Programs</h4>
                  <p className={styles.featureText}>
                    Microfinance, skills development, and entrepreneurship training to build sustainable livelihoods.
                  </p>
                  <Link to="/campaigns" className={styles.featureLink}>
                    Learn More &rarr;
                  </Link>
                </div>
              </div>
            </div>
            <div className="col-lg-4 mb-4">
              <div className={`card h-100 ${styles.featureCard}`}>
                <div className="card-body p-4">
                  <i className={`fas fa-hands-helping ${styles.featureIcon}`}></i>
                  <h4 className={styles.featureTitle}>Emergency Relief</h4>
                  <p className={styles.featureText}>
                    Rapid response to natural disasters, food security, and essential support for vulnerable communities.
                  </p>
                  <Link to="/campaigns" className={styles.featureLink}>
                    Learn More &rarr;
                  </Link>
                </div>
              </div>
            </div>
            <div className="col-lg-4 mb-4">
              <div className={`card h-100 ${styles.featureCard}`}>
                <div className="card-body p-4">
                  <i className={`fas fa-graduation-cap ${styles.featureIcon}`}></i>
                  <h4 className={styles.featureTitle}>Education & Training</h4>
                  <p className={styles.featureText}>
                    Comprehensive education programs, vocational training, and digital literacy initiatives.
                  </p>
                  <Link to="/campaigns" className={styles.featureLink}>
                    Learn More &rarr;
                  </Link>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>

      {/* Final Call to Action Section */}
      <section className={styles.finalCtaSection}>
        <div className="container text-center">
          <h2 className={styles.finalCtaTitle}>Ready to Make a Difference?</h2>
          <p className="lead mb-4">
            Join thousands of donors and volunteers who are building a better future for Bangladesh.
          </p>
          <div className={styles.finalCtaButtons}>
            <Link to="/donate" className={styles.ctaFinalPrimary}>
              <i className="fas fa-heart me-2"></i>
              Start Donating
            </Link>
            <Link to="/volunteer-activity" className={styles.ctaFinalSecondary}>
              <i className="fas fa-hands-helping me-2"></i>
              Become a Volunteer
            </Link>
            <Link to="/request-campaign" className={styles.ctaFinalSecondary}>
              <i className="fas fa-plus-circle me-2"></i>
              Request a New Campaign
            </Link>
          </div>
        </div>
      </section>
    </div>
  );
};

export default HomePage;

