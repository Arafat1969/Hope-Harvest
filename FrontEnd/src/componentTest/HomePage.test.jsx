import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import HomePage from "../components/HomePage";

describe("HomePage Component", () => {
  it("renders hero section with title", () => {
    render(
      <MemoryRouter>
        <HomePage />
      </MemoryRouter>
    );
    expect(screen.getByText("Hope Harvest")).toBeInTheDocument();
    expect(
      screen.getByText("Sowing Seeds of Change, Reaping a Brighter Future.")
    ).toBeInTheDocument();
  });

  it("displays call-to-action buttons", () => {
    render(
      <MemoryRouter>
        <HomePage />
      </MemoryRouter>
    );
    expect(screen.getByText("Donate Now")).toBeInTheDocument();
    expect(screen.getByText("View Campaigns")).toBeInTheDocument();
  });

  it("shows statistics section", () => {
    render(
      <MemoryRouter>
        <HomePage />
      </MemoryRouter>
    );
    expect(screen.getByText("Lives Impacted")).toBeInTheDocument();
    expect(screen.getByText("Active Campaigns")).toBeInTheDocument();
    expect(screen.getByText("Taka Raised")).toBeInTheDocument();
    expect(screen.getByText("Volunteers")).toBeInTheDocument();
  });

  it("displays feature cards", () => {
    render(
      <MemoryRouter>
        <HomePage />
      </MemoryRouter>
    );
    expect(screen.getByText("Self-Reliance Programs")).toBeInTheDocument();
    expect(screen.getByText("Emergency Relief")).toBeInTheDocument();
    expect(screen.getByText("Education & Training")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <HomePage />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});