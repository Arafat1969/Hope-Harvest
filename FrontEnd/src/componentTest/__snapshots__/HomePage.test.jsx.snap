// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <section
      class="heroSection"
    >
      <div
        class="heroBackground"
      >
        <img
          alt="Hope Harvest initiative background"
          class="backgroundImage visible"
          src="https://upload.wikimedia.org/wikipedia/commons/thumb/c/c9/Agriculture_in_Rural_Bangladesh.jpg/2560px-Agriculture_in_Rural_Bangladesh.jpg"
        />
        <img
          alt="Hope Harvest initiative background"
          class="backgroundImage "
          src="https://tds-images.thedailystar.net/sites/default/files/styles/very_big_201/public/images/2023/03/22/char_1.jpg"
        />
        <img
          alt="Hope Harvest initiative background"
          class="backgroundImage "
          src="https://cdn.pixabay.com/photo/2021/02/12/16/08/children-6008967_1280.jpg"
        />
        <img
          alt="Hope Harvest initiative background"
          class="backgroundImage "
          src="https://s.france24.com/media/display/cb41bca0-6201-11ef-9d3e-005056bf30b7/w:1280/p:16x9/5db8797c814b2f749a6d17aa7a5dc022e7f7be13.jpg"
        />
        <img
          alt="Hope Harvest initiative background"
          class="backgroundImage "
          src="https://im.indiatimes.in/amp/2019/Jul/trees_1562664811.jpg?w=1200&h=900&cc=1&webp=1&q=75"
        />
      </div>
      <div
        class="container heroContent"
      >
        <div
          class="row"
        >
          <div
            class="col-lg-7 text-center text-lg-start"
          >
            <h1
              class="heroTitle"
            >
              Hope Harvest
            </h1>
            <h2
              class="heroSubtitle"
            >
              Sowing Seeds of Change, Reaping a Brighter Future.
            </h2>
            <p
              class="heroText"
            >
              We empower communities across Bangladesh through sustainable microfinance, education, and direct relief, creating lasting, positive change from the ground up.
            </p>
            <div
              class="d-flex gap-3 justify-content-center justify-content-lg-start"
            >
              <a
                class="btn ctaButtonPrimary"
                href="/donate"
              >
                <i
                  class="fas fa-hand-holding-heart me-2"
                />
                Donate Now
              </a>
              <a
                class="btn ctaButtonSecondary"
                href="/campaigns"
              >
                View Campaigns
              </a>
            </div>
          </div>
        </div>
      </div>
    </section>
    <section
      class="statsSection"
    >
      <div
        class="container"
      >
        <div
          class="row g-4"
        >
          <div
            class="col-lg-3 col-md-6"
          >
            <div
              class="statCard"
            >
              <i
                class="fas fa-users statIcon"
              />
              <h3
                class="statNumber"
              >
                <span>
                  0+
                </span>
              </h3>
              <p
                class="statLabel"
              >
                Lives Impacted
              </p>
            </div>
          </div>
          <div
            class="col-lg-3 col-md-6"
          >
            <div
              class="statCard"
            >
              <i
                class="fas fa-bullhorn statIcon"
              />
              <h3
                class="statNumber"
              >
                <span>
                  0+
                </span>
              </h3>
              <p
                class="statLabel"
              >
                Active Campaigns
              </p>
            </div>
          </div>
          <div
            class="col-lg-3 col-md-6"
          >
            <div
              class="statCard"
            >
              <i
                class="fas fa-hand-holding-heart statIcon"
              />
              <h3
                class="statNumber"
              >
                <span>
                  0+
                </span>
              </h3>
              <p
                class="statLabel"
              >
                Taka Raised
              </p>
            </div>
          </div>
          <div
            class="col-lg-3 col-md-6"
          >
            <div
              class="statCard"
            >
              <i
                class="fas fa-hands-helping statIcon"
              />
              <h3
                class="statNumber"
              >
                <span>
                  0+
                </span>
              </h3>
              <p
                class="statLabel"
              >
                Volunteers
              </p>
            </div>
          </div>
        </div>
      </div>
    </section>
    <section
      class="featuresSection"
    >
      <div
        class="container"
      >
        <div
          class="text-center mb-5"
        >
          <h2
            class="sectionTitle"
          >
            How We Make a Difference
          </h2>
          <p
            class="sectionSubtitle"
          >
            Our comprehensive approach to community development
          </p>
        </div>
        <div
          class="row"
        >
          <div
            class="col-lg-4 mb-4"
          >
            <div
              class="card h-100 featureCard"
            >
              <div
                class="card-body p-4"
              >
                <i
                  class="fas fa-seedling featureIcon"
                />
                <h4
                  class="featureTitle"
                >
                  Self-Reliance Programs
                </h4>
                <p
                  class="featureText"
                >
                  Microfinance, skills development, and entrepreneurship training to build sustainable livelihoods.
                </p>
                <a
                  class="featureLink"
                  href="/campaigns"
                >
                  Learn More →
                </a>
              </div>
            </div>
          </div>
          <div
            class="col-lg-4 mb-4"
          >
            <div
              class="card h-100 featureCard"
            >
              <div
                class="card-body p-4"
              >
                <i
                  class="fas fa-hands-helping featureIcon"
                />
                <h4
                  class="featureTitle"
                >
                  Emergency Relief
                </h4>
                <p
                  class="featureText"
                >
                  Rapid response to natural disasters, food security, and essential support for vulnerable communities.
                </p>
                <a
                  class="featureLink"
                  href="/campaigns"
                >
                  Learn More →
                </a>
              </div>
            </div>
          </div>
          <div
            class="col-lg-4 mb-4"
          >
            <div
              class="card h-100 featureCard"
            >
              <div
                class="card-body p-4"
              >
                <i
                  class="fas fa-graduation-cap featureIcon"
                />
                <h4
                  class="featureTitle"
                >
                  Education & Training
                </h4>
                <p
                  class="featureText"
                >
                  Comprehensive education programs, vocational training, and digital literacy initiatives.
                </p>
                <a
                  class="featureLink"
                  href="/campaigns"
                >
                  Learn More →
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
    <section
      class="finalCtaSection"
    >
      <div
        class="container text-center"
      >
        <h2
          class="finalCtaTitle"
        >
          Ready to Make a Difference?
        </h2>
        <p
          class="lead mb-4"
        >
          Join thousands of donors and volunteers who are building a better future for Bangladesh.
        </p>
        <div
          class="finalCtaButtons"
        >
          <a
            class="ctaFinalPrimary"
            href="/donate"
          >
            <i
              class="fas fa-heart me-2"
            />
            Start Donating
          </a>
          <a
            class="ctaFinalSecondary"
            href="/volunteer-activity"
          >
            <i
              class="fas fa-hands-helping me-2"
            />
            Become a Volunteer
          </a>
          <a
            class="ctaFinalSecondary"
            href="/request-campaign"
          >
            <i
              class="fas fa-plus-circle me-2"
            />
            Request a New Campaign
          </a>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
